### PR TITLE
Setup Arrow-Meta with arrow-gradle-config

### PR DESCRIPTION
**Pull request by i-walker**
- automatic semantic versioning
- binaryCompability tooling
- api files are being updated for each PR
- generate tag and generate alpha tag github workflows
- publishing alpha/beta/rc tags and final too
This PR will produce a 1.6.1-alpha.1 release. Since I pushed a 1.6.0 tag. Which will be cleaned up after

### CODE REVIEW

**Review by raulraja — APPROVED** [review]
Thanks @i-walker !, I'm also tagging @JavierSegoviaCordoba, he has more experience with this setup more than I do.

File: .github/workflows/generate-alpha-tag.yaml [hunk 1]
```diff
@@ -0,0 +1,59 @@
+name: generate-alpha-tag
+
+on:
+  push:
+    branches:
+      - main
+
+env:
+  GRADLE_OPTS: -Dorg.gradle.daemon=false -Dorg.gradle.parallel=false -Dorg.gradle.jvmargs="-Xmx5g -XX:+UseParallelGC -XX:+HeapDumpOnOutOfMemoryError -Dfile.encoding=UTF-8"
```
Comment (JavierSegoviaCordoba) [inline .github/workflows/generate-alpha-tag.yaml:9]: meta needs no daemon and no parallel to work too?

File: .gitignore [hunk 2]
```diff
@@ -16,6 +16,8 @@ bin/
 gen/
 out/
 
+kotlin-js-store
```
Comment (JavierSegoviaCordoba) [inline .gitignore:19]: I think this should be in the repo and not ignored based on comments in Slack from JB engineers.

**Review by JavierSegoviaCordoba — APPROVED** [review]
Good job :) A few I leave a few doubts

File: .gitignore (hunk 2 above)
Comment (i-walker) [reply .gitignore:19]: I am also in favor. Adding the kotlin-js-store hasn't been added consistently to all arrow projects. 
Maybe it makes sense to decide, whether we should have it in all projects or not
cc @raulraja , @nomisRev

Comment (nomisRev) [reply .gitignore:19]: > based on comments in Slack from JB engineers.

If it's officially recommended to keep this file in git then I think we should.

Comment (i-walker) [reply .gitignore:19]: Will add it :D and do the same for the proofs and analysis plugin

File: .github/workflows/generate-alpha-tag.yaml (hunk 1 above)
Comment (i-walker) [reply .github/workflows/generate-alpha-tag.yaml:9]: as in our chat, we could take a look if we can improve this in another PR

Comment (i-walker) [reply .github/workflows/generate-alpha-tag.yaml:9]: as in our chat, we could take a look if we can improve this in another PR

**Pull request merged**